src/jackknife.jl

```diff
@@ -0,0 +1,45 @@
+"""
+Use bootweights to creaye replicate weights using Rao-Wu bootstrap. The function accepts a `SurveyDesign` and returns a `ReplicateDesign` which has additional columns for replicate weights. 
+
+```jldoctest
+julia> using Random
+
+julia> dclus1 = SurveyDesign(apiclus1; clusters = :dnum, popsize=:fpc);
+```
+"""
+function jknweights(design::SurveyDesign; replicates = 10, rng = MersenneTwister(1234))
+    stratified = groupby(design.data, design.strata)
+    H = length(keys(stratified))
+    substrata_dfs = DataFrame[]
+    for h = 1:H
+        substrata = DataFrame(stratified[h])
+        cluster_sorted = sort(substrata, design.cluster)
+        psus = unique(cluster_sorted[!, design.cluster])
+        npsus = [(count(==(i), cluster_sorted[!, design.cluster])) for i in psus]
+        nh = length(psus)
+        cluster_weights = cluster_sorted[!, design.weights]
+        for replicate = 1:replicates
+            randinds = rand(rng, 1:(nh), (nh - 1))
+            cluster_sorted[!, "replicate_"*string(replicate)] =
+                vcat(
+                    [
+                        fill((nh - count(==(i), randinds)) * ((nh - 1) / nh), npsus[i]) for

```

Comment:
check this change from section 2.1 theory. in Rao-wu bootstrap the `nh -1` term isnt there. looks promising, but check what is JKn correction factor

Reply:
you mean it should be something like this `nh / (nh - 1)`?